monailabel/transform/post.py

```diff
@@ -37,23 +37,52 @@
 
 
 # TODO:: Move to MONAI ??
-
-
 class LargestCCd(MapTransform):
     def __init__(self, keys: KeysCollection, has_channel: bool = True):
+        """
+        MapTransform that applies the get_largest_cc function to label images in a data dictionary.
+
+        Args:
+            keys: keys of the data dictionary to be transformed.
+            has_channel: whether the data has channel dimension or not.
+        """
         super().__init__(keys)
         self.has_channel = has_channel
 
     @staticmethod
     def get_largest_cc(label):
-        largest_cc = np.zeros(shape=label.shape, dtype=label.dtype)
-        for i, item in enumerate(label):
-            item = measure.label(item, connectivity=1)
-            if item.max() != 0:
-                largest_cc[i, ...] = item == (np.argmax(np.bincount(item.flat)[1:]) + 1)
-        return largest_cc
+        """
+        Returns the largest connected component (LCC) of each label image in the input array.
+
+        Args:
+            label: a numpy array of label images with shape (num_images, height, width[, channels]).
+
+        Returns:
+            A numpy array of LCC images with the same shape as the input array.
+        """
+        largest_cc = np.zeros_like(label)
+        for item in label:
+            try:
+                item_label = ndimage.label(item, connectivity=1)
+                if item_label.max() != 0:
+                    item_lcc = item_label == ndimage.mode(item_label, axis=None)[0][0]
+                else:
+                    item_lcc = np.zeros_like(item)
+            except ValueError:
+                item_lcc = np.zeros_like(item)
+            largest_cc.append(item_lcc)
```

Comment:
largest_cc = np.zeros_like(label) this will already have the initial shape.. and now you are trying to append..

Can you confirm if you have verified the result... (testing it via 3D Slicer for Deepgrow Pipeline for which this class is currently used)